app/pages/lab/export-workflows-dialog.jsx

```diff
@@ -36,17 +43,30 @@ class ExportWorkflowsDialog extends React.Component {
     getWorkflowsInOrder(project, { fields: 'display_name' })
       .then((workflows) => {
         this.setState({ workflows, loading: false });
-      });
-  }
 
-  toggleExport() {
-    if (this.workflowList.selectedIndex >= 0) {
-      this.setState({ workflowSelected: true });
-    }
+        workflows.forEach((workflow) => {
+          workflow.get('classifications_export')
+            .then((media) => {
+              const mediaState = this.state.media;
+              mediaState[workflow.id] = media[0];
+              this.setState({ media: mediaState });
+            })
+            .catch((error) => {
+              if (error.status !== 404) {
+                this.props.onFail(error);
+              } else {
+                const workflowErrorState = this.state.workflowError;
```

Comment:
This is the error handling that should be done instead of `onFail` for the get request for errors that are not 404. This should be moved into the true block of the if statement.